File: core-banking-prototype-laravel-main/core-banking-prototype-laravel-main/sdks/php/src/Resources/Transactions.php
<?php

namespace FinAegis\Resources;

use FinAegis\Models\PaginatedResponse;
use FinAegis\Models\Transaction;

class Transactions extends BaseResource
{
    /**
     * Get transaction details.
     *
     * @param  string  $transactionId  Transaction UUID
     */
    public function get(string $transactionId): Transaction
    {
        $response = $this->get("/transactions/{$transactionId}");

        return new Transaction($response['data']);
    }

    /**
     * List transactions.
     *
     * @param  int  $page  Page number
     * @param  int  $perPage  Items per page
     */
    public function list(int $page = 1, int $perPage = 20): PaginatedResponse
    {
        $response = $this->get('/transactions', ['page' => $page, 'per_page' => $perPage]);

        return new PaginatedResponse($response, Transaction::class);
    }
}

File: core-banking-prototype-laravel-main/core-banking-prototype-laravel-main/sdks/python/finaegis/resources/transactions.py
"""
Transactions resource for the FinAegis SDK
"""

from ..types import Transaction, PaginatedResponse
from .base import BaseResource


class TransactionsResource(BaseResource):
    """Manage transactions in the FinAegis platform."""
    
    def list(self, page: int = 1, per_page: int = 20) -> PaginatedResponse:
        """
        List all transactions.
        
        Args:
            page: Page number
            per_page: Items per page
            
        Returns:
            PaginatedResponse containing Transaction objects
        """
        response = self._get('/transactions', params={'page': page, 'per_page': per_page})
        return PaginatedResponse.from_dict(response, Transaction)
    
    def get(self, transaction_id: str) -> Transaction:
        """
        Get transaction details.
        
        Args:
            transaction_id: Transaction ID
            
        Returns:
            Transaction object
        """
        response = self._get(f'/transactions/{transaction_id}')
        return Transaction.from_dict(response['data'])